Metalama.Framework.Engine/Pipeline/DesignTime/DesignTimeSyntaxTreeGenerator.cs

```diff
@@ -153,6 +172,120 @@ void ProcessTransformationsOnType( IGrouping<INamedType, ITransformation> transf
+                            SyntaxKind.ThisConstructorInitializer,
+                            ArgumentList(
+                                SeparatedList( 
+                                    initialParameters.SelectAsArray( 
+                                        p => 
+                                            Argument( 
+                                                p.DefaultValue != null
+                                                    ? NameColon( p.Name )
+                                                    : null, 
+                                                GetArgumentRefToken( p ), 
+                                                IdentifierName( p.Name ) ) ) ) ) ),
+                        Block() ) );
+
+                if ( initialConstructor.Parameters.Any( p => p.DefaultValue != null ) )
+                {
+                    // Target constructor has optional parameters.
+                    // If there is no constructor without optional parameters, we need to generate it to avoid ambiguous match.
+
+                    var nonOptionalParameters = initialParameters.Where( p => p.DefaultValue == null ).ToArray();
+                    var optionalParameters = initialParameters.Where( p => p.DefaultValue != null ).ToArray();
+
+                    if ( existingSignatures.Add( nonOptionalParameters.SelectAsArray( p => ((ISymbol) p.Type.GetSymbol(), p.RefKind) ) ) )
+                    {
+                        constructors.Add(
+                            ConstructorDeclaration(
+                                List<AttributeListSyntax>(),
+                                initialConstructor.GetSyntaxModifierList(),
+                                Identifier( initialConstructor.DeclaringType.Name ),
+                                syntaxGenerationContext.SyntaxGenerator.ParameterList( nonOptionalParameters, initialCompilationModel ),
+                                ConstructorInitializer(
+                                    SyntaxKind.ThisConstructorInitializer,
+                                    ArgumentList(
+                                        SeparatedList( 
+                                            nonOptionalParameters.SelectAsArray( p => Argument( null, GetArgumentRefToken( p ), IdentifierName( p.Name ) ) ) )
+                                        .AddRange(
+                                            optionalParameters.SelectAsArray(
+                                                p =>
+                                                    Argument(
+                                                        NameColon( p.Name ),
+                                                        GetArgumentRefToken(p),
+                                                        LiteralExpression(
+                                                            SyntaxKind.DefaultLiteralExpression,
+                                                            Token( SyntaxKind.DefaultKeyword ) ) ) ) ) ) ),
```

Comment:
Will this work correctly for the edge case where the initial constructor has both optional parameters and `params` (which doesn't have a `DefaultValue`, but has to be last)?

Reply:
Good point, this is not supported even in compile-time pipeline so I skipped it. There was no bug though.